person.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'nameable'
+
+class Person < Nameable
+  def initialize(name, age = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def correct_name
```

Comment:
**You have done great job!**

- [ ] However, your based on requirement testing if the app is working as expected is not completely held up. When I run the test in `person.rb`:

```ruby
person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name
``` 

it shows me errors. Kindly fix the issue and resubmit your app. If you have any question please reach me out via slack